hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -924,128 +698,110 @@ def event_to_string(self, event):
         return result
 
 
-def _draw_predecessor(rnd, xs):
-    r = bytearray()
-    any_strict = False
-    for x in to_bytes_sequence(xs):
-        if not any_strict:
-            c = rnd.randint(0, x)
-            if c < x:
-                any_strict = True
-        else:
-            c = rnd.randint(0, 255)
-        r.append(c)
-    return hbytes(r)
-
-
-def _draw_successor(rnd, xs):
-    r = bytearray()
-    any_strict = False
-    for x in to_bytes_sequence(xs):
-        if not any_strict:
-            c = rnd.randint(x, 255)
-            if c > x:
-                any_strict = True
+class GenerationParameters(object):
+    """Parameters to control generation of examples."""
+
+    AVERAGE_ALPHABET_SIZE = 3
+
+    ALPHABET_FACTOR = math.log(1.0 - 1.0 / AVERAGE_ALPHABET_SIZE)
+
+    def __init__(self, random):
+        self.__random = random
+        self.__zero_chance = None
+        self.__max_chance = None
+        self.__pure_chance = None
+        self.__alphabet = {}
+
+    def draw_bytes(self, n):
+        """Draw an n-byte block from the distribution defined by this instance
+        of generation parameters."""
+        alphabet = self.alphabet(n)
+
+        if alphabet is None:
+            return uniform(self.__random, n)
+
+        return self.__draw_without_alphabet(n)
+
+    def __draw_without_alphabet(self, n):
```

Comment:
The name of this method seems inconsistent with the fact that it does indeed use the alphabet.

(I’m guessing that’s why the alphabet generator gets in trouble for calling it.)

Reply:
Yeah, that's because I messed up entirely in tidyig up some code. 😳  The correct version of it should be pushed now.